### PR TITLE
Added Go challenge.

### PR DESCRIPTION
**Pull request by vishnu-deepsource**

### CODE REVIEW

File: golang.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Password managers have become an essential utility in day-to-day life.  However, as developers, using a browser is, at times, not an option.  Imagine you have ssh'd into a server and need to access credentials to your database.  Your job is to build a simple headless password manager that helps you access your credentials over a simple TCP link.
+
+### User Stories
+
+Feature | Desciption
+------- | ----------
+**TCP Based Protocol** | Users should be able to interact with your password manager over a simple TCP connection using a simple text based protocol.
+**User Management** | A new user should be able to sign up with a master password using a `signup <password>` command.  The user should be able to login and logout of a session using simple commands.
+**Password Management** | The user should be able to create, retrieve, update and delete passwords.  Each password can have a label associated with it.  Look above for an example of the command.
```
Comment (sanket-deepsource) [inline golang.md:20]:
```suggestion
**Password Management** | The user should be able to create, retrieve, update and delete passwords.  Each password can have a label associated with it.  Look below for an example of the command.
```

File: golang.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Go Developer Challenge
+
+* **Timebox** 4 hours
+* **Technologies** Golang
```
Comment (sanket-deepsource) [inline golang.md:4]:
```suggestion
* **Technologies:** Go
```

File: golang.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Go Developer Challenge
+
+* **Timebox** 4 hours
```
Comment (sanket-deepsource) [inline golang.md:3]:
```suggestion
* **Timebox:** 4 hours
```

File: golang.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Pick one of these challenges to get started.  Good luck! ✨
+
+## Challenge #1: Password Manager CLI
+
+Password managers have become an essential utility in day-to-day life.  However, as developers, using a browser is, at times, not an option.  Imagine you have ssh'd into a server and need to access credentials to your database.  Your job is to build a simple headless password manager that helps you access your credentials over a simple TCP link.
+
+### User Stories
+
+Feature | Desciption
```
Comment (sanket-deepsource) [inline golang.md:16]:
```suggestion
Feature | Description
```

File: golang.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+------- | ----------
+**TCP Based Protocol** | Users should be able to interact with your password manager over a simple TCP connection using a simple text based protocol.
```
Comment (sanket-deepsource) [inline golang.md:18]:
```suggestion
**TCP-based Protocol** | Users should be able to interact with your password manager over a simple TCP connection using a simple text-based protocol.
```

File: golang.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Go Developer Challenge
+
+* **Timebox** 4 hours
+* **Technologies** Golang
+* **Tests:** Must have
+* **Documentation:** Nice to have
+
+Pick one of these challenges to get started.  Good luck! ✨
+
+## Challenge #1: Password Manager CLI
```
Comment (sanket-deepsource) [inline golang.md:10]: So essentially, the person needs to write both a server and a CLI client?

**Review by sanket-deepsource — CHANGES_REQUESTED** [review]

File: golang.md (hunk 6 above)
Comment (vishnu-deepsource) [reply golang.md:10]: No.  Just the server.  We will use telnet to issue commands.  I'll add that.

**Review by sanket-deepsource — APPROVED** [review]

**Pull request merged**